Framework/DataHandling/src/LoadBinStl.cpp

```diff
@@ -0,0 +1,105 @@
+#include "MantidDataHandling/LoadBinStl.h"
+#include "MantidAPI/FileProperty.h"
+#include "MantidGeometry/Objects/MeshObject.h"
+#include "MantidKernel/BinaryStreamReader.h"
+#include <Poco/File.h>
+#include <fstream>
+#include <iostream>
+
+namespace Mantid {
+namespace DataHandling {
+
+bool LoadBinStl::isBinarySTL() {
+  // each triangle is 50 bytes
+  const uint32_t SIZE_OF_TRIANGLE = 50;
+  Poco::File stlFile = Poco::File(m_filename);
+  auto fileSize = stlFile.getSize();
+  if (fileSize < 84) {
+    // File is smaller than header plus number of triangles, cannot be binary
+    // format stl
+    return false;
+  }
+  uint32_t numberTrianglesLong;
+  std::ifstream myFile(m_filename.c_str(), std::ios::in | std::ios::binary);
+  Kernel::BinaryStreamReader streamReader = Kernel::BinaryStreamReader(myFile);
+  numberTrianglesLong = getNumberTriangles(streamReader);
+  myFile.close();
+  if (!(fileSize == (84 + (numberTrianglesLong * SIZE_OF_TRIANGLE)))) {
+    // File is not the Header plus the number of triangles it claims to be long,
+    // invalid binary Stl
+    return false;
+  }
+  // if both conditions pass, file is likely binary stl
+  return true;
+}
+
+uint32_t
+LoadBinStl::getNumberTriangles(Kernel::BinaryStreamReader streamReader) {
+  uint32_t numberTrianglesLong;
+  // skip header
+  streamReader.moveStreamToPosition(80);
+  // Read the number of triangles
+  streamReader >> numberTrianglesLong;
+  return numberTrianglesLong;
+}
+
+std::unique_ptr<Geometry::MeshObject> LoadBinStl::readStl() {
+  std::ifstream myFile(m_filename.c_str(), std::ios::in | std::ios::binary);
+  uint32_t numberTrianglesLong;
+  Kernel::BinaryStreamReader streamReader = Kernel::BinaryStreamReader(myFile);
+  numberTrianglesLong = getNumberTriangles(streamReader);
+  uint32_t next = 96;
+  const int STEPSIZE = 50;
```

Comment:
These variable names could maybe be a bit better, i.e. what is 50 bytes long that is being stepped over.